client/ayon_core/hosts/houdini/plugins/create/create_hda.py

```diff
@@ -68,13 +87,20 @@ def create_instance_node(
         else:
             hda_node = to_hda
 
-        hda_node.setName(node_name)
+        # If user tries to create the same HDA instance more than
+        # once, then all of them will have the same product name and
+        # point to the same hda_file_name. But, their node names will
+        # be incremented.
+        hda_node.setName(node_name, unique_name=True)
         self.customize_node_look(hda_node)
         return hda_node
 
     def create(self, product_name, instance_data, pre_create_data):
         instance_data.pop("active", None)
 
+        self.min_num_inputs = pre_create_data["min_num_inputs"]
+        self.max_num_inputs = pre_create_data["max_num_inputs"]
```

Comment:
It seems the HDA creator is the only creator in Houdini currently that overrides `create_instance_node` method. As such, I'd actually instead change its signature and have it just pass along `pre_create_data` directly to the `create_instance_node` method since it's just much better design than storing it as attributes on the Creator plug-in class (which is also shared between instances, etc.)

So basically, just add `pre_create_data` as argument to `create_instance_node` method.

Reply:
Does that mean I need to add `pre_create_data` as in the following ? 

1. on calling `self.create_instance_node` inside super(..).create(..)
https://github.com/ynput/ayon-core/blob/df976a4ab8900b14c70331519afd563e871c8d09/client/ayon_core/hosts/houdini/api/plugin.py#L186-L187

2. in `super(..).create_instance_node(..)`
https://github.com/ynput/ayon-core/blob/df976a4ab8900b14c70331519afd563e871c8d09/client/ayon_core/hosts/houdini/api/plugin.py#L143-L145

3. last but not least in `CreateHDA.create_instance_node`

---
I'm a little bit frustrated because `create_instance_node` is called from `super(..).create(..)` not from `CreateHDA.craete(..)` 

Also, `super(..).create(..)` doesn't support `instance_node` as argument

Reply:
Exactly, see for example: https://github.com/BigRoy/ayon-core/commit/6e68802c7617e1aba635be89d60051583e1226ff

You could merge that branch into yours to get the commit (or cherry pick the commit via git)

Reply:
I cherry-picked it, Thanks!